rfcs/45-stageless.md

```diff
@@ -2,488 +2,457 @@
-  - e.g. `.add_system(physics.in_set(GameSets::Physics)`
+fn main() {
+    App::new()
+        /* ... */
+        // helper function that registers state machine resource
+        // and adds OnEnter and OnExit sets for its states
+        .add_states([GameState::Playing, GameState::Paused])
+        .add_set(
+            // systems and sets have the same API for scheduling
+            GameSet::Menu
+                // sets can go inside other sets
+                .in_set(CoreSet::Update)
+                // sets can have conditions, these must be true for the systems inside to run
+                .run_if(state_equals(GameState::Paused))
+        )
+        .add_system(
+            some_system
+                .in_set(GameSet::Menu)
+                // systems and sets can have multiple conditions (they must all be true)
+                .run_if(some_condition_system)
+                .run_if(|value: Res<Value>| value > 9000)
+        )
+        // multiple systems and sets can be added at once
+        .add_many(
+          // there are macros available for describing a linear sequence of systems and sets
+          chain![
+            GameSet::SubMenu,
+            // process commands using a dedicated system
+            apply_buffers.named("FlushMenu"),
+          ]
+          // configure all nodes in the chain at once
+          .in_set(GameSet::Menu)
+          .after(some_system)
+        )
+        /* ... */
+        .run();
+}
+```
 
-This configuration is additive: adding another ordering constraint, run criteria or set does not replace the existing configuration.
+### Configuring run order
 
-System configuration can be stored in a `SystemConfig` struct.
-This can be useful to reuse, compose and quickly apply complex configuration strategies, before applying these strategies to sets or individual systems.
+The simplest way users can configure systems is by saying *when* they should run using the `.before`, `.after`, and `.in_set` methods. These constraints are relative to another system or set. Run order constraints involving sets are eventually flattened into constraints between the individual systems. Any constraints the user specifies are used to derive dependency graphs, which (along with the data access) tells the engine where systems can run in parallel. The only limit is that every graph must be **solvable**. If any relationship constraint cannot be met (e.g. A was configured to run both before and after system B), it will result in an error.
 
-### System set configuration
+Bevy provides the `App::add_many` method for configuring multiple systems and sets at once. Using `nodes![a, b, c, ...].chain()` (or the `chain![a, b, c, ...]` shortcut) will create dependencies between the successive elements. (Note: This is different from "system chaining" in Bevy 0.7 and earlier. That concept has been renamed to "system piping" to avoid overlap.)

```

Comment:
```suggestion
Bevy provides the `App::add_many` method for configuring multiple systems and sets at once. Using `nodes![a, b, c, ...].chain()` (or the `chain![a, b, c, ...]` shortcut) will create dependencies between the successive elements.
(Note: This is different from "system chaining" in Bevy 0.7 and earlier. That concept has been renamed to "system piping" to avoid overlap.)
```